core/src/main/java/io/substrait/relation/Project.java

```diff
@@ -8,7 +8,7 @@
 import org.immutables.value.Value;
 
 @Value.Immutable
-public abstract class Project extends SingleInputRel {
+public abstract class Project extends SingleInputRel implements HasExtension {
```

Comment:
Why not just make `Rel extends HasExtension`?

Reply:
`ExtensionSingle`, `ExtensionLeaf` and `ExtensionMulti` all ultimately inherent from `Rel` as well, however they don't have extensions associated with them. 

I originally put `getRelExtension` directly in `Rel`, but moved it into the `HasExtension` to work around this.

Reply:
Hmm.. not strongly held but if we adjusted per above the registry/list above, the method for extension rels would simply return the common extension in those cases (and not both) since the common extension is independent of specific extension.